packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -126,7 +126,11 @@ const DropdownMenuTrigger = React.forwardRef<DropdownMenuTriggerElement, Dropdow
           })}
           onKeyDown={composeEventHandlers(props.onKeyDown, (event) => {
             if (disabled) return;
-            if (['Enter', ' '].includes(event.key)) context.onOpenToggle();
+            if (['Enter', ' '].includes(event.key)) {
+              context.onOpenToggle();
+              // prevent passing keydown event to the focused MenuItem element in Firefox
+              if (!context.open) event.preventDefault();
+            }
             if (event.key === 'ArrowDown') context.onOpenChange(true);
             // prevent keypresses from scrolling window
             if ([' ', 'ArrowDown'].includes(event.key)) event.preventDefault();
```

Comment:
I think this would be enough no?
I don't think we need the extra `if (!context.open)`

```suggestion
            if (['Enter', ' '].includes(event.key)) context.onOpenToggle();
            if (event.key === 'ArrowDown') context.onOpenChange(true);
            // prevent keydown from scrolling window / first focused item to execute
            // that keydown (inadvertently closing the menu)
            if (['Enter', ' ', 'ArrowDown'].includes(event.key)) event.preventDefault();
```

Reply:
Yes, looks like you are right, we can get rid of unnecessary check for `open` menu